File: my-folder/0021-merge-two-sorted-lists/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummyHead = ListNode(0,None)
        temp = dummyHead

        while l1 is not None and l2 is not None:
            newNode = ListNode(0,None)

            if l1.val <= l2.val:
                newNode.val = l1.val
                l1 = l1.next if l1 is not None else None
            elif l2 is not None:
                newNode.val = l2.val
                l2 = l2.next if l2 is not None else None
            
            temp.next = newNode
            temp = newNode
        
        while l1 is not None:
            newNode = ListNode(l1.val,None)
            temp.next = newNode
            temp = newNode
            l1 = l1.next if l1 is not None else None
        
        while l2 is not None:
            newNode = ListNode(l2.val,None)
            temp.next = newNode
            temp = newNode
            l2 = l2.next if l2 is not None else None
        


        
        return dummyHead.next

File: my-folder/0021-merge-two-sorted-lists/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* DummyHead = new ListNode();
        ListNode* temp = DummyHead;

        while (list1 && list2){
            ListNode* newNode = new ListNode();
            temp->next = newNode;
            if (list1->val <= list2->val){
                newNode->val = list1->val;
                list1 = list1->next;
            }
            else {
                newNode->val = list2->val;
                list2 = list2->next;
            }
            temp = newNode;
        }

        while (list1){
            ListNode* newNode = new ListNode();
            temp->next = newNode;
            newNode->val = list1->val;
            temp = newNode;
            list1 = list1->next;
        }

        while (list2){
            ListNode* newNode = new ListNode();
            temp->next = newNode;
            newNode->val = list2->val;
            temp = newNode;
            list2 = list2->next;
        }
        return DummyHead->next;
    }
};
